From 0491c6873a0342d876530c97e4b66d0f7764fc09 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Tue, 12 Apr 2022 17:37:37 +0800
Subject: [PATCH] gstreamer1.0-libav: add caps field to skip h265 parser [1/1]

PD#SWPL-76043

Problem:
H265parser recaculate the pts
whill will make decoder output wrong pts value

Solution:
After ffmpeg, the data has been parsed
So there is no need to re-parse data
add caps field to skip h265 parser

Verify:
AP222
---
 ext/libav/gstavcodecmap.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/libav/gstavcodecmap.c b/ext/libav/gstavcodecmap.c
index 6db76f0..3c18ee8 100755
--- a/ext/libav/gstavcodecmap.c
+++ b/ext/libav/gstavcodecmap.c
@@ -1305,7 +1305,9 @@ gst_ffmpeg_codecid_to_caps (enum AVCodecID codec_id,
     case AV_CODEC_ID_HEVC:
       caps =
           gst_ff_vid_caps_new (context, NULL, codec_id, encode, "video/x-h265",
-          "alignment", G_TYPE_STRING, "au", NULL);
+          "parsed", G_TYPE_BOOLEAN, "true",
+          "alignment", G_TYPE_STRING, "au", 
+          "stream-format", G_TYPE_STRING, "byte-stream", NULL);
       if (!encode) {
         GValue arr = { 0, };
         GValue item = { 0, };
-- 
2.25.1

